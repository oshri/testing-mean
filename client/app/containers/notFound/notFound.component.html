<div fxFill>
	<div class="message" fxLayout="column" fxLayoutAlign="center center">
		<p>"Sorry ! the page you are looking for can't be found"</p>
		<button class="back-btn" mat-raised-button color="accent" routerLink="/">Back to home</button>
	</div>
</div>
  