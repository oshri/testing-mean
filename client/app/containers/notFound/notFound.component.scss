:host(mean-not-found) {
	.message {
		position: absolute;
		top: 50%;
		left: 50%;
		transform: translate(-50%, -50%);

		p {
			color: white;
		}

	}
}
