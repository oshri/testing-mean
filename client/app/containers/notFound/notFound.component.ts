import { Component } from '@angular/core';

@Component({
	selector: 'mean-not-found',
	styleUrls: ['./notFound.component.scss'],
	templateUrl: './notFound.component.html'
})
export class NotFoundComponent {}
