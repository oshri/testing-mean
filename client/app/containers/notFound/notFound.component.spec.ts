import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA, CUSTOM_ELEMENTS_SCHEMA, DebugElement } from '@angular/core';
import { By } from '@angular/platform-browser';
import { NotFoundComponent } from './notFound.component';

describe('NotFoundComponent', () => {
	let component: NotFoundComponent;
	let fixture: ComponentFixture<NotFoundComponent>;
	let messageContainer: DebugElement;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [NotFoundComponent]
		}).compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(NotFoundComponent);
		component = fixture.componentInstance;
		messageContainer = fixture.debugElement.query(By.css('.message'));

		fixture.detectChanges();
	});

	it('should compile', () => {
		expect(fixture).toMatchSnapshot();
	});

	it('should create the NotFoundComponent', async(() => {
		const notFound = fixture.debugElement.componentInstance;
		expect(notFound).toBeTruthy();
	}));

	it('should render 404 description', async(() => {
		const messageDe = messageContainer.query(By.css('p'));
		const p: HTMLElement = messageDe.nativeElement;
		expect(p.textContent).toContain("Sorry ! the page you are looking for can't be found");
	}));
});
