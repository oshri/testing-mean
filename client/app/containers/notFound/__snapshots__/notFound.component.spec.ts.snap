// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundComponent should compile 1`] = `
<mean-not-found>
  <div
    fxfill=""
  >
    <div
      class="message"
      fxlayout="column"
      fxlayoutalign="center center"
    >
      <p>
        "Sorry ! the page you are looking for can't be found"
      </p>
      <button
        class="back-btn"
        color="accent"
        mat-raised-button=""
        routerlink="/"
      >
        Back to home
      </button>
    </div>
  </div>
</mean-not-found>
`;
